### PR TITLE
FIX report for section names having only digits

### PR DESCRIPTION
**Pull request by mainakjas**
closes https://github.com/mne-tools/mne-python/issues/1887

cc @dengemann

### CODE REVIEW

File: mne/report.py [hunk 1]
```diff
@@ -1497,10 +1497,9 @@ def _clean_varnames(s):
     # Remove invalid characters
     s = re.sub('[^0-9a-zA-Z_]', '', s)
 
-    # Remove leading characters until we find a letter or underscore
-    s = re.sub('^[^a-zA-Z_]+', '', s)
-
-    return s
+    # add report_ at the beginning so that the javascript class names
+    # are valid ones
+    return 'report_' + s
```
Comment (agramfort) [inline mne/report.py]: KISS :)

**Comment by dengemann** [comment]
There is some bug Mainak.

```
======================================================================
ERROR: Test adding html str to mne report.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/testenv/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/miniconda/envs/testenv/lib/python2.7/site-packages/numpy/testing/decorators.py", line 146, in skipper_func
    return f(*args, **kwargs)
  File "/home/travis/miniconda/envs/testenv/lib/python2.7/site-packages/numpy/testing/decorators.py", line 146, in skipper_func
    return f(*args, **kwargs)
  File "/home/travis/miniconda/envs/testenv/lib/python2.7/site-packages/mne-0.9.dev0-py2.7.egg/mne/tests/test_report.py", line 210, in test_add_htmls_to_section
    idx = report._sectionlabels.index(section)
ValueError: 'html_section' is not in list
```

**Comment by mainakjas** [comment]
woops! Let me take a look.

**Comment by mainakjas** [comment]
should be fixed now

**Comment by Eric89GXL** [comment]
Should we add tests to make sure it's not missed in future changes?

**Comment by dengemann** [comment]
> Should we add tests to make sure it's not missed in future changes?

One simple test would be to check that section names are as expected. On master the cleaning function would generate empty strings as section names (--> disaster).

**Comment by Eric89GXL** [comment]
@mainakjas don't forget about this one, too -- we proabbly also want it for 0.9 :)

**Comment by agramfort** [comment]
merge if done and happy

**Comment by mainakjas** [comment]
I need to add a test. I'll do that a bit later.

**Comment by Eric89GXL** [comment]
I have changed my mind, this is probably reasonably enough tested already. I didn't realize @dengemann's error actually arose from a test already.

**Comment by Eric89GXL** [comment]
Thanks for the fix!

**Pull request merged**